collect_app/src/main/java/org/odk/collect/android/forms/DatabaseFormRepository.java

```diff
@@ -0,0 +1,51 @@
+package org.odk.collect.android.forms;
+
+import android.database.Cursor;
+
+import org.odk.collect.android.dao.FormsDao;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+public class DatabaseFormRepository implements FormRepository {
+
+    @Override
+    public boolean contains(String jrFormID) {
+        try (Cursor cursor = new FormsDao().getFormsCursorForFormId(jrFormID)) {
+            return cursor != null && cursor.getCount() > 0;
+        }
+    }
+
+    @Override
+    public List<Form> getAll() {
```

Comment:
In our live review we talked about in a follow-up PR considering changing this to a `Set`. There's no reason for the results to be ordered and sets will make some client code cleaner. We also talked about replacing `contains` with a get by jrFormId (which should also return a set with potentially multiple versions). No action needed now but writing it down so we don't forget.